dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java

```diff
@@ -30,6 +30,8 @@
      */
     private static final String CHECK_NUMBER = "^-?\\d+(\\.\\d+)?$";
 
+    private static final String LINUX_USERNAME_PATTERN = "[a-z_][a-z\\d_]{0,30}";
```

Comment:
This regular expression pattern is not correct. `[a-zA-Z0-9_.][a-zA-Z0-9_.-]*[$]?` should be used by `man useradd` and it has better compatibility.

![image](https://user-images.githubusercontent.com/4902714/111784824-cdbd4e00-88f6-11eb-8806-4c879755ed2a.png)

Reply:
![image](https://user-images.githubusercontent.com/21101605/111860431-5d5c0e80-8982-11eb-8547-9af6881359a6.png)

This is the constraints of ubuntu OS above. different Linux OS have different constraints of username.  
We can give the strictest rules to fit all of those different OS. which refers to https://stackoverflow.com/questions/6949667/what-are-the-real-rules-for-linux-usernames-on-centos-6-and-rhel-6

Reply:
make sense